lib/Service/ApiService.php

```diff
@@ -113,12 +113,12 @@ public function create($fileId = null, $filePath = null, $token = null, $guestNa
 			$readOnly = $this->documentService->isReadOnly($file, $token);
 
 			$this->sessionService->removeInactiveSessions($file->getId());
-			$activeSessions = $this->sessionService->getActiveSessions($file->getId());
+			$remainingSessions = $this->sessionService->getAllSessions($file->getId());
 			$freshSession = false;
-			if ($forceRecreate || count($activeSessions) === 0) {
+			if ($forceRecreate || count($remainingSessions) === 0) {
+				$freshSession = true;
 				try {
 					$this->documentService->resetDocument($file->getId(), $forceRecreate);
-					$freshSession = true;
```

Comment:
You're right - that's exactly what this change is aiming for. Let me explain the reasoning in more detail.

We would not need the session to apply the steps - they are self contained - as long as all of them are present.

However if the session does not exist anymore it must have been removed. If steps are still present this only happens during a close call. So close was called on the session.

If other sessions are still around we won't run into this block anyway. If there were other sessions around during the close they would have been closed as well and would have cleaned up the steps. Therefore if we end up here the session must have been the last active session when it was closed. In that case it will have cleaned up it's steps and the remaining unsaved changes are just leftovers from the race condition and yes... we cannot apply them anymore because we lack the history leading up to them.